src/error.rs

```
@@ -26,6 +26,14 @@ pub enum Error {
     #[error("illegal argument: {0}")]
     IllegalArgument(String),
 
+    /// Generic error for HTTP connections.
+    #[error("http: {0}")]
+    Http(String),
```

Why not propagate the error from the `http` crate?

---

http::Error and hyper::Error do not implement PartialEq or Eq

---

Do we still need `PartialEq` and `Eq` if we're using `assert_matches`? Are there other places we need equality?